### PR TITLE
Inform sync explicitly about new best block

### PR DESCRIPTION
**Pull request by bkchr**
Instead of "fishing" the new best block out of the processed blocks, we
now tell sync directly that there is a new best block. It also makes
sure that we update the corresponding sync handshake to the new best
block. This is required for parachains as they first import blocks and
declare the new best block after being made aware of it by the relay chain.

### CODE REVIEW

**Review by tomaka — APPROVED** [review]
Looks good to me, but I'm not an expert in the sync code

File: client/network/src/protocol.rs [hunk 1]
```diff
@@ -528,10 +528,9 @@ impl<B: BlockT, H: ExHashT> Protocol<B, H> {
 		self.sync.num_sync_requests()
 	}
 
-	/// Sync local state with the blockchain state.
-	pub fn update_chain(&mut self) {
-		let info = self.context_data.chain.info();
-		self.sync.update_chain_info(&info.best_hash, info.best_number);
+	/// Inform sync about new best imported block.
+	pub fn new_best_block_imported(&mut self, hash: B::Hash, number: NumberFor<B>) {
+		self.sync.update_chain_info(&hash, number);
```
Comment (arkpar) [inline client/network/src/protocol.rs:533]: Should probably also call `self.behaviour.set_notif_protocol_handshake` here, @tomaka?

Comment (bkchr) [reply client/network/src/protocol.rs:533]: I'm doing it. Just look down :D

Comment (tomaka) [reply client/network/src/protocol.rs:533]: I think it's called right below, just not shown in the diff.

**Review by arkpar — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

**Pull request merged**